cmd/net/src/lib.rs

```diff
@@ -473,6 +503,89 @@ fn net_counters(context: &mut ExecutionContext) -> Result<()> {
+
+    let mut ksz_tx = [0; 3];
+    let mut ksz_rx = [0; 3];
+    for port in 0..3 {
+        let k_tx = k_tx[port].as_struct()?;
+        let k_rx = k_rx[port].as_struct()?;
+        for t in ["unicast", "broadcast", "multicast"] {
+            ksz_tx[port] += value(k_tx, t);
+            ksz_rx[port] += value(k_rx, t);
+        }
+    }
+
+    let v_tx = s["vsc85x2_tx"].as_array()?;
+    let v_rx = s["vsc85x2_rx"].as_array()?;
+    let v_mac_valid = s["vsc85x2_mac_valid"].as_base()?.as_bool().unwrap();
+    let value = |v: &Struct, s: &str| v[s].as_base().unwrap().as_u16().unwrap();
+
+    let mut v_mac_tx = [0; 2];
+    let mut v_mac_rx = [0; 2];
+    let mut v_media_tx = [0; 2];
+    let mut v_media_rx = [0; 2];
+    for port in 0..2 {
+        let v_tx = v_tx[port].as_struct()?;
+        let v_rx = v_rx[port].as_struct()?;
+        if v_mac_valid {
+            v_mac_tx[port] = value(v_tx, "mac_good");
+            v_mac_rx[port] = value(v_rx, "mac_good");
+        }
+        v_media_tx[port] = value(v_tx, "media_good");
+        v_media_rx[port] = value(v_rx, "media_good");
+    }
+
+    let mac = |i: u16| {
+        if v_mac_valid {
+            i.to_string().green()
+        } else {
+            "--".to_string().dimmed()
+        }
+    };
+
+    println!(
+        "            ┌──────────────────┐            ┌───────────────────┐
+            │ {}          │            │ {}           │
+            │                  │tx        rx│                   │tx
+            │            {:>6}├───────────►│{:<6}       {:>6}├───────►
+            │                  │1          0│                   │0
+┌────┐    rx│            {:>6}│◄───────────┤{:<6}       {:>6}│◄───────
+│    ├─────►│{:<6}            │rx        tx│                   │rx
+│ SP │     3│                  │            │                   │
+│    │◄─────┤{:<6}            │tx        rx│                   │tx
+└────┘    tx│            {:>6}├───────────►│{:<6}       {:>6}├───────►
+            │                  │2          1│                   │1
+            │            {:>6}│◄───────────┤{:<6}       {:>6}│◄───────
+            │                  │rx        tx│                   │rx
+            └──────────────────┘            └───────────────────┘
+                                             {}           {}
+    ",
+        "KSZ8463".bold(),
+        "VSC85x2".bold(),
+        ksz_tx[0].to_string().green(),
+        v_media_rx[0].to_string().green(),
+        mac(v_mac_tx[0]),
+        ksz_rx[0].to_string().green(),
+        v_media_tx[0].to_string().green(),
+        mac(v_mac_rx[0]),
+        ksz_rx[2].to_string().green(),
+        ksz_tx[2].to_string().green(),
+        ksz_tx[1].to_string().green(),
+        v_media_rx[1].to_string().green(),
+        mac(v_mac_tx[1]),
+        ksz_rx[1].to_string().green(),
+        v_media_tx[1].to_string().green(),
+        mac(v_mac_rx[1]),
+        "MEDIA".dimmed(),
+        "MAC".dimmed()
```

Comment:
nit, take it or leave it: this might be a lot easier to figure out if we used named parameters in the format placeholders? OTTOH the Unicode art would be less well aligned, so I dunno...